frontend/extension/content.js

```diff
@@ -1,32 +1,31 @@
-// Copyright 2022 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 // TODO: Configure TypeScript for the content scripts
 function addstylesheet(filename) {
-    var link = document.createElement('link')
-    link.href = chrome.runtime.getURL(filename)
-    link.type = 'text/css'
-    link.rel = 'stylesheet'
-    document.getElementsByTagName('head')[0].appendChild(link)
+    var link = document.createElement('link');
+    link.href = chrome.runtime.getURL(filename);
+    link.type = 'text/css';
+    link.rel = 'stylesheet';
+    document.getElementsByTagName('head')[0].appendChild(link);
+}
+
+addstylesheet('styles.css');
+
+function getPageText() {
+    let res = [];
+    const allText = Array.from(document.body.getElementsByTagName('p')).map(
+        (e) => e.innerHTML
+    );
+
+    for (const text of allText) {
+        const span_regex = /<span.*>/g;
+        const a_tag_regex = /<\/?a([^>])*>/g;
+
+        if (text.charAt(0) !== '<' && !span_regex.exec(text)) {
+            res.push(text.replaceAll(a_tag_regex, ''));
+        }
```

Comment:
Can we refactor the logic "removing tag" here in another function so that it doesn't get too long if we want to add more conditions to it in the future?

Reply:
yeah this PR is merging into Saurav's branch. we should address this after we merge since it's his code block too